js/dropdown.js

```diff
@@ -0,0 +1,251 @@
+define([
+    './dropdownOption'
+], function(DropDownOption) {
+
+    var DropDown = Backbone.View.extend({
+
+        initialize: function(settings) {
+            _.bindAll(this, 'onStartInteraction', 'onButtonClick', 'onListBlur', 'onKeyDown');
+            this.settings = _.defaults(settings, this.getDefaults());
+            this.placeholder = null;
+            this.options = [];
+            this.setUpElements();
+            this.setUpItems();
+            this.select(this.settings.value);
+            this.addEventListeners();
+            this.toggleOpen(false);
+            this.settings.load.call(this, this);
+            this.trigger('load', this);
+        },
+
+        getDefaults: function() {
+            return {
+                load: DropDown.defaults.load,
+                openList: DropDown.defaults.openList,
+                closeList: DropDown.defaults.closeList,
+                scrollToItem: DropDown.defaults.scrollToItem
+            };
+        },
+
+        setUpElements: function() {
+            this.$list = this.$('ul');
+            this.$button = this.$('button');
+            this.$inner = this.$button.find('.dropdown__inner');
+            this.$input = this.$('input');
+        },
+
+        setUpItems: function() {
+            var $options = this.$('li');
+            $options.each(function(index, el) {
+                var option = new DropDownOption({
+                    parent: this,
+                    el: el
+                });
+                if (option.isPlaceholder()) {
+                    this.placeholder = option;
+                    return;
+                }
+                this.options.push(option);
+            }.bind(this));
+        },
+
+        addEventListeners: function() {
+            this.$button.on({
+                'mousedown touchstart': this.onStartInteraction,
+                click: this.onButtonClick
+            });
+            this.$list.on('blur', this.onListBlur);
+            $(document).on('keydown', this.onKeyDown);
+        },
+
+        onStartInteraction: function() {
+            this.wasOpen = this.isOpen();
+        },
+
+        deselectAll: function() {
+            this.placeholder.deselect();
+            this.options.forEach(function(option) {
+                option.deselect();
+            });
+        },
+
+        onButtonClick: function(event) {
+            if (this.wasOpen || this.isOpen()) {
+                // click toggle list:
+                // if the list is open and the button is clicked
+                // the blur event will fire first closing the list.
+                // this code is to prevent the list from reopening
+                // in that situation
+                return this.wasOpen = false;
```

Comment:
it's very minor but changing this to `return (this.wasOpen = false);` would stop code linters like JSHint from flagging this with a 'Did you mean to return a conditional instead of an assignment?' warning.

Reply:
Please put `return;` onto its own line.